peers/app_request_network.go

```diff
@@ -126,95 +133,155 @@ func NewNetwork(
 		return nil, err
 	}
 
-	validatorClient := validators.NewCanonicalValidatorClient(logger, cfg.GetPChainAPI())
+	for _, peer := range manuallyTrackedPeers {
+		logger.Info(
+			"Manually Tracking peer (startup)",
+			zap.String("ID", peer.ID.String()),
+			zap.String("IP", peer.PublicIP.String()),
+		)
+		testNetwork.ManuallyTrack(peer.ID, peer.PublicIP)
+	}
 
-	arNetwork := &appRequestNetwork{
-		network:              testNetwork,
-		handler:              handler,
-		infoAPI:              infoAPI,
-		logger:               logger,
-		lock:                 new(sync.Mutex),
-		validatorClient:      validatorClient,
-		metrics:              metrics,
-		manuallyTrackedPeers: manuallyTrackedPeers,
+	// Connect to a sample of the primary network validators
+	peers, err := infoAPI.Peers(context.Background(), nil)
+	if err != nil {
+		logger.Error(
+			"Failed to get peers",
+			zap.Error(err),
+		)
+		return nil, err
 	}
+	pClient := platformvm.NewClient(cfg.GetPChainAPI().BaseURL)
+	numConnected := 0
+	for _, peer := range peers {
+		vdrs, err := pClient.GetCurrentValidators(context.Background(), constants.PrimaryNetworkID, []ids.NodeID{peer.ID})
+		if err != nil {
+			panic(err)
+		}
+		// Only track the peer if it is a primary network validator
+		if len(vdrs) == 0 {
+			continue
+		}
+		logger.Info(
+			"Manually tracking bootstrap node",
+			zap.String("ID", peer.ID.String()),
+			zap.String("IP", peer.PublicIP.String()),
+		)
+		testNetwork.ManuallyTrack(peer.ID, peer.PublicIP)
+		numConnected++
+		if numConnected >= NumBootstrapNodes {
+			break
+		}
+	}
+
 	go logger.RecoverAndPanic(func() {
 		testNetwork.Dispatch()
 	})
 
+	arNetwork := &appRequestNetwork{
+		network:         testNetwork,
+		handler:         handler,
+		infoAPI:         infoAPI,
+		logger:          logger,
+		lock:            new(sync.RWMutex),
+		validatorClient: validatorClient,
+		metrics:         metrics,
+		trackedSubnets:  trackedSubnets,
+		manager:         manager,
+	}
+
+	arNetwork.startUpdateValidators()
+
 	return arNetwork, nil
 }
 
-// ConnectPeers connects the network to peers with the given nodeIDs.
-// Returns the set of nodeIDs that were successfully connected to.
-func (n *appRequestNetwork) ConnectPeers(nodeIDs set.Set[ids.NodeID]) set.Set[ids.NodeID] {
+func (n *appRequestNetwork) TrackSubnet(subnetID ids.ID) {
+	n.lock.RLock()
+	if n.trackedSubnets.Contains(subnetID) {
+		n.lock.RUnlock()
+		return
+	}
+	n.lock.RUnlock()
+
 	n.lock.Lock()
 	defer n.lock.Unlock()
+	n.logger.Debug("Tracking subnet", zap.String("subnetID", subnetID.String()))
+	n.trackedSubnets.Add(subnetID)
+	updateValidatorSet(context.Background(), n.logger, subnetID, n.manager, n.validatorClient)
```

Comment:
Is it a common pattern to hold the lock outside of the function that performs the update?

Reply:
Yeah, it might be better to make `updateValidatorSet` a receiver on the `network`? And then we just pass in the `subnetID` to the function.

Reply:
Moved the lock management inside of `updateValidatorSet` and made it a `network` receiver